sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-saveAssessmentData.e2e-spec.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  createE2EDataSources,
+  E2EDataSources,
+  saveFakeApplication,
+} from "@sims/test-utils";
+import { ICustomHeaders, IOutputVariables } from "zeebe-node";
+import {
+  createFakeWorkerJob,
+  FAKE_WORKER_JOB_RESULT_PROPERTY,
+  MockedZeebeJobResult,
+} from "../../../../../test/utils/worker-job-mock";
+import { createTestingAppModule } from "../../../../../test/helpers";
+import { AssessmentController } from "../../assessment.controller";
+import { SaveAssessmentDataJobInDTO } from "../../assessment.dto";
+import { createFakeSaveAssessmentDataPayload } from "./save-assessment-data-factory";
+
+describe("AssessmentController(e2e)-saveAssessmentData", () => {
+  let db: E2EDataSources;
+  let assessmentController: AssessmentController;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    db = createE2EDataSources(dataSource);
+    assessmentController = nestApplication.get(AssessmentController);
+  });
+
+  it("Should save student assessment data when it had not been updated yet.", async () => {
+    // Arrange
+    const savedApplication = await saveFakeApplication(db.dataSource);
+    const assessmentData = { totalAssessmentNeed: 9999 };
+    const saveAssessmentDataPayload = createFakeSaveAssessmentDataPayload(
+      savedApplication.currentAssessment.id,
+      assessmentData,
+    );
+
+    // Act
+    const result = await assessmentController.saveAssessmentData(
```

Comment:
`createFakeSaveAssessmentDataPayload` itself is returning a fake job.

![image](https://github.com/bcgov/SIMS/assets/54600590/bf633b8a-4a5d-4373-96ce-cde67d329071)

do we need to create again a fake job here?

what about 
![image](https://github.com/bcgov/SIMS/assets/54600590/314d8512-52aa-4649-91b0-96df1b4b2ea7)